node/network/gossip-support/src/lib.rs

```diff
@@ -256,13 +256,14 @@ where
 					}
 				}
 
-				// Gossip topology is only relevant for authorities in the current session.
-				let our_index =
-					ensure_i_am_an_authority(&self.keystore, &session_info.discovery_keys).await?;
-
 				if is_new_session {
 					self.update_authority_status_metrics(&session_info).await;
 
+					// Gossip topology is only relevant for authorities in the current session.
+					let our_index =
+						ensure_i_am_an_authority(&self.keystore, &session_info.discovery_keys)
```

Comment:
instead of calling this function twice, can we return the result of it in update_authority_status_metrics?